components/pkg-export-docker/src/lib.rs

```diff
@@ -179,3 +183,41 @@ pub fn export(ui: &mut UI, build_spec: BuildSpec, naming: &Naming) -> Result<Doc
 
     Ok(image)
 }
+
+/// Creates a build specification and naming policy from Cli arguments, and then exports a Docker
+/// image to a Docker engine from them.
+///
+/// # Errors
+///
+/// * If the actual import fails.
+/// * Current directory does not exist.
+/// * There are insufficient permissions to access the current directory.
+/// * If pushing the image to remote registry fails.
+/// * Parsing of credentials fails.
+/// * The image (tags) cannot be removed.
+pub fn export_for_cli_matches(ui: &mut UI, matches: &clap::ArgMatches) -> Result<()> {
+    let default_channel = channel::default();
+    let default_url = hurl::default_bldr_url();
+    let spec = BuildSpec::new_from_cli_matches(&matches, &default_channel, &default_url);
+    let naming = Naming::new_from_cli_matches(&matches);
+
+    let docker_image = export(ui, spec, &naming)?;
+    docker_image.create_report(
+        ui,
+        env::current_dir()?.join("results"),
+    )?;
+
+    if matches.is_present("PUSH_IMAGE") {
+        let credentials = Credentials::new(
+            naming.registry_type,
+            matches.value_of("REGISTRY_USERNAME").unwrap(),
```

Comment:
Using `expect` might be a bit nicer for the user here and below.

Reply:
actually it shouldn't never get a NONE here so unwrap() is fine. The dependencies setup with clap ensure that this arg is always specified if PUSH_IMAGE is.

Reply:
but I do like the idea of using expect() instead, just to be super safe. That'd be independent of these changes though.

Reply:
Pushed two additional commits and address this. I believe I've addressed all your comments now but let me know if I missed something.